pkgs/applications/science/logic/z3/default.nix

```diff
@@ -35,6 +35,9 @@ let common = { version, sha256, patches ? [ ] }:
       ++ optionals ocamlBindings [ ocaml findlib zarith ]
     ;
     propagatedBuildInputs = [ python.pkgs.setuptools ];
+    passthru.optional-dependencies = {
+      python = [ python.pkgs.setuptools ];
+    };
```

Comment:
passthru.optional-dependencies should only be used for extras that can be installed via ``pip install abc[extra]``.

Reply:
Good to know. I switched to use `z3.requiredPythonModules`, set in `toPythonModule` to the `propagatedBuildInputs` of the Z3 derivation.